sdk/core/azure-core/src/test/java/com/azure/core/implementation/jackson/FlattenDangling.java

```diff
@@ -22,4 +27,52 @@ public FlattenDangling setFlattenedProperty(String flattenedProperty) {
         this.flattenedProperty = flattenedProperty;
         return this;
     }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) {
+        jsonWriter.writeStartObject();
+
+        if (flattenedProperty != null) {
+            jsonWriter.writeFieldName("a")
+                .writeStartObject()
+                .writeFieldName("flattened")
+                .writeStartObject()
+                .writeStringField("property", flattenedProperty)
+                .writeEndObject()
+                .writeEndObject();
+        }
+
+        return jsonWriter.writeEndObject().flush();
+    }
+
+    public static FlattenDangling fromJson(JsonReader jsonReader) {
+        return JsonUtils.readObject(jsonReader, (reader, token) -> {
+            String flattenedProperty = null;
+
+            while (reader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = reader.getFieldName();
+                token = reader.nextToken();
+
+                if ("a".equals(fieldName) && token == JsonToken.START_OBJECT) {
```

Comment:
> && token == JsonToken.START_OBJECT

what if it's not the case?
It's the same concern as before on common place for validation + failure + logging and finding a pattern that would help models be consistent.